### ISSUE TITLE
精调运行失败 ninja

### ISSUE THREAD

**Issue by yangxingrui** [ymcui/Chinese-LLaMA-Alpaca#596]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
- [X] 由于相关依赖频繁更新，请确保按照[Wiki](https://github.com/ymcui/Chinese-LLaMA-Alpaca/wiki)中的相关步骤执行
- [X] 我已阅读[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)、[LlamaChat](https://github.com/alexrozanski/LlamaChat)等，同时建议到对应的项目中查找解决方案
- [X] 模型正确性检查：务必检查模型的[SHA256.md](https://github.com/ymcui/Chinese-LLaMA-Alpaca/blob/main/SHA256.md)，模型不对的情况下无法保证效果和正常运行

### 问题类型

模型训练与精调

### 基础模型

Alpaca-Plus-7B

### 操作系统

Linux

### 详细描述问题

模型合并之后推理效果良好，模型对中文的理解能力可以接受。但对特定领域知识微调过程中报错，数据集格式示例：
{"title":"修改后的立法法全文公布","content":"新华社受权于18日全文播发修改后的《中华人民共和国立法法》，修改后的立法法分为“总则”“法律”“行政法规”“地方性法规、自治条例和单行条例、规章”“适用与备案审查”“附则”等6章，共计105条。"}，读取数据集部分代码已在build_dataset.py中修改。
运行指令：run_sft.sh
lr=1e-4
lora_rank=8
lora_alpha=32
lora_trainable="q_proj,v_proj,k_proj,o_proj,gate_proj,down_proj,up_proj"
modules_to_save="embed_tokens,lm_head"
lora_dropout=0.05

pretrained_model='/public/home/yangxingr/Chinese-LLaMA-Alpaca-main/combined'
chinese_tokenizer_path='/public/home/yangxingr/Chinese-LLaMA-Alpaca-main/combined'
dataset_dir='/public/home/yangxingr/Chinese-LLaMA-Alpaca-main/ft_data/'
per_device_train_batch_size=1
per_device_eval_batch_size=1
training_steps=100
gradient_accumulation_steps=1
output_dir='/public/home/yangxingr/Chinese-LLaMA-Alpaca-main/sft'
validation_file='/public/home/yangxingr/Chinese-LLaMA-Alpaca-main/data/lcsts_data_small_test.json'
RANDOM=1000
deepspeed_config_file=ds_zero2_no_offload.json

CUDA_VISIBLE_DEVICES=0 torchrun --nnodes 1 --nproc_per_node 1 run_clm_sft_with_peft.py \
    --deepspeed ${deepspeed_config_file} \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --validation_split_percentage 0.001 \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --per_device_eval_batch_size ${per_device_eval_batch_size} \
    --do_train \
    --do_eval \
    --seed $RANDOM \
    --fp16 \
    --max_steps ${training_steps} \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.03 \
    --weight_decay 0 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 20 \
    --evaluation_strategy steps \
    --eval_steps 250 \
    --save_steps 500 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --max_seq_length 512 \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --lora_rank ${lora_rank} \
    --lora_alpha ${lora_alpha} \
    --trainable ${lora_trainable} \
    --lora_dropout ${lora_dropout} \
    --torch_dtype float16 \
    --validation_file ${validation_file} \
    --ddp_find_unused_parameters False

### 依赖情况（代码类问题务必提供）

transformers              4.28.1
peft                      0.3.0.dev0
torch                     1.13.1
python                    3.9.0
ninja                     1.11.1
huggingface-hub           0.15.1
datasets                  2.2.0
deepspeed                 0.9.4

### 运行日志或截图

报错部分：
![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca/assets/95137682/04cf32e8-aa18-447f-b1eb-d7bf240f5adb)

**Comment by zhang2514yuchi** [comment]
试试加这个
```
conda install https://anaconda.org/brown-data-science/gc
c/5.4.0/download/linux-64/gcc-5.4.0-0.tar.bz2
```

**Comment by yangxingrui** [comment]
非常棒，这有效，感谢你的回复。
